ExchangeRateApp/Controllers/CurrencyPairViewController.swift

```diff
@@ -7,116 +7,72 @@
-                let data = rawData
-                currency.append(CurrencyCodeData(code: data, fullName: "", flag: UIImage(named: "EUR")!))
+        title = "Currency Pair"
+        
+        if let path = Bundle.main.path(forResource: "Currencies", ofType: "json"),
+            let dataArray = try? JSONSerialization.jsonObject(
+                with: Data(contentsOf: URL(fileURLWithPath: path))
+            ) as? [String] {
+            
+            for data in dataArray {
+                currencies.append(CurrencyCodeData(code: data))
             }
         }
     }
-   
-    
-
-// MARK: UITableViewDataSource
-
-func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
     
-    return currency.count
-}
-
-    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+    // MARK: UITableViewDataSource
     
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        currencies.count
+    }
     
-    guard let cell = tableView.dequeueReusableCell(withIdentifier: "currencyCell", for: indexPath) as? CurrencyTableViewCell else {
-        return UITableViewCell()}
-        let item = currency[indexPath.row]
-   
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: "currencyCell", for: indexPath) as? CurrencyTableViewCell else {
+            return UITableViewCell()
+        }
+        
+        let item = currencies[indexPath.row]
+        
         cell.currencyAbriviationLabel?.text = item.code
         cell.countryLabel?.text = item.fullName.map { $0.rawValue }
         cell.countryFlagImageView.image = item.flag
         cell.countryFlagImageView.layer.cornerRadius = cell.countryFlagImageView.frame.height / 2
-        cell.countryFlagImageView.layer.masksToBounds = false
         cell.countryFlagImageView.clipsToBounds = true
         cell.countryFlagImageView.layer.borderWidth = 1
-    return cell
-
-
+        
+        return cell
     }
-
+    
     func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
-        let item = currency[indexPath.row]
-        isTapped = true
-        selectedCellPath = indexPath
-        if isTapped {
-            print("First Currency Pair Selected Cell:", indexPath)
-            print("First Content:", item.code)
-            array1.append(Code(code: item.code, fullName: item.fullName!.rawValue, currency: ""))
+        let item = currencies[indexPath.row]
+        
+        print("First Currency Pair Selected Cell:", indexPath)
+        print("First Content:", item.code)
+        
+        delegate?.pickedCurrency(item)
+        
+        selectedCurrencies.append(item)
+        
+        if selectedCurrencies.count == 1 {
+            currencies.remove(at: indexPath.row)
             tableView.reloadData()
-            print(array1 )
```

Comment:
Too many `reloadData` calls, you need to reload data only when it changed (where _it_ is data that is returned in table view data source methods)